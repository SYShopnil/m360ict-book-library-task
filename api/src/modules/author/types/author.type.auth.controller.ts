export interface IAuthorLoginBOdy {
  password: string;
  email: string;
}
export interface IPayloadDataOfAccessToken {
  id: number;
  email: string;
}
