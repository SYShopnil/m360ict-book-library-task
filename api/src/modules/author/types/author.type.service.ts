import { IAuthor } from '../../../type/entity';

export interface IGetAllAuthors {
  authors: IAuthor[];
  total: number;
  totalPages: number;
}

export interface IRGetAuthorById extends IAuthor {
  email: string;
}
export interface IRGetAuthorByEmail extends IAuthor {
  email: string;
  password: string;
}
