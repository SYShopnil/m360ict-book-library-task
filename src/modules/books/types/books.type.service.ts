import { IBook } from '../../../type/entity';

export interface GetBooksOptions {
  page: number;
  pageSize: number;
  title?: string;
  author?: string;
}

export interface IRGetBookById extends IBook {}
export interface IRGetBooksByAuthor extends IBook {
  author_name: string;
  authors_bio: string;
}
