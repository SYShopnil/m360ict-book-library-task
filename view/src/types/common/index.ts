export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EAuth {
  AuthTokenCookieName = "auth",
}

export enum EDataTestId {
  cLoginFormWithSubmit = "cLoginFormWithSubmit",
  cLogoutContainer = "cLogoutContainer",
  rUserList = "rUserList",
  SHeaderMain = "SHeaderMain",
  SProductSection = "SProductSection",
  SProfile = "SProfile",
  CPaginationTrack = "CPaginationTrack",
  RSingleProduct = "RSingleProduct",
  RTableBody = "RTableBody",
  SProductCard = "SProductCard",
  CProductSearchBarContainer = "CProductSearchBarContainer",
}
