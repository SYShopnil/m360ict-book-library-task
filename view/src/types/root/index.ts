export type { IIconStore } from "./_icon";
export { IconName as IconEnums } from "./_icon";
export type { IButton, INormalButton, IRedirectButton } from "./Button";
export { BtnColorSchema } from "./Button";
