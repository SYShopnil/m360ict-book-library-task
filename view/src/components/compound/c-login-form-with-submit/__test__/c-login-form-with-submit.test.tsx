import { render, screen } from "@testing-library/react";
import { CLoginFormWithSubmit } from "..";
import { EDataTestId } from "@src/types/common";

describe("When Clone Form with submit rendered ", () => {
  it("Expect it should render successfully", async () => {
    const { getByTestId } = render(<CLoginFormWithSubmit />); // ARRANGE
    // const myElement = getByTestId(EDataTestId.cLoginFormWithSubmit);
    const myElement = screen.getByTestId(EDataTestId.cLoginFormWithSubmit);

    expect(myElement).toMatchSnapshot();
  });
});
