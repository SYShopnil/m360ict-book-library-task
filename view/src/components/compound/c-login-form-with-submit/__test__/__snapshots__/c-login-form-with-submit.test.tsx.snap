// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Clone Form with submit rendered  Expect it should render successfully 1`] = `
<div
  class="flex justify-center items-center h-[80vh] bg-gray-100"
  data-testid="cLoginFormWithSubmit"
>
  <form
    class="bg-white p-8 rounded shadow-md"
  >
    <h2
      class="text-2xl font-semibold mb-4 text-center"
    >
      Login
    </h2>
    <div
      class="mb-4"
    >
      <label
        class="block text-gray-700"
        for="email"
      >
        Email:
      </label>
      <input
        class="w-full border rounded px-3 py-2 mt-1 focus:outline-none focus:ring focus:border-blue-300"
        id="email"
        name="email"
        required=""
        type="email"
        value=""
      />
    </div>
    <div
      class="mb-6"
    >
      <label
        class="block text-gray-700"
        for="password"
      >
        Password:
      </label>
      <input
        class="w-full border rounded px-3 py-2 mt-1 focus:outline-none focus:ring focus:border-blue-300"
        id="password"
        name="password"
        required=""
        type="password"
        value=""
      />
    </div>
    <span
      class="cursor-pointer"
    >
      <span
        class="flex justify-center items-center  "
      >
        <span
          class="text-base md:text-base_2  flex justify-center items-center px-[1rem] py-[0.5rem] rounded-[0.563rem] w-[100%] h-[100%] bg-[#EBF5EC] hover:bg-[#7F4D4F]  duration-[0.5s] text-[#7F4D4F] hover:text-[#EBF5EC]   border-[0.063rem] hover:border-[#EBF5EC] border-[#7F4D4F]   "
        >
          <span>
            Login
          </span>
        </span>
      </span>
    </span>
    <span
      class="text-bold "
    />
  </form>
</div>
`;
